### PR TITLE
#754: Use Modello for XML reports (also fixes #755)

### LINKED ISSUE #755: Regression: XML reports are serialized incorrectly
While looking at XML renderers, I've noticed that they're now rendered incorrectly. Please hold the release. I'm fixing it.
```xml
<PluginUpdatesReport>
	<summary>
		<usingLastVersion>1</usingLastVersion>
		<nextVersionAvailable>1</nextVersionAvailable>
		<nextIncrementalAvailable>0</nextIncrementalAvailable>
		<nextMinorAvailable>10</nextMinorAvailable>
		<nextMajorAvailable>1</nextMajorAvailable>
	</summary>
	<pluginManagements>
		<pluginManagement>
			<groupId>org.apache.maven.plugins</groupId>
			<artifactId>maven-antrun-plugin</artifactId>
			<currentVersion>1.3</currentVersion>
			<nextVersion>1.4</nextVersion>
			<minors>
				<optional[minor]>1.4</optional[minor]>
				<optional[minor]>1.5</optional[minor]>
				<optional[minor]>1.6</optional[minor]>
				<optional[minor]>1.7</optional[minor]>
				<optional[minor]>1.8</optional[minor]>
			</minors>
			<majors>
				<optional[major]>3.0.0</optional[major]>
				<optional[major]>3.1.0</optional[major]>
			</majors>
			<status>minor available</status>
		</pluginManagement>
...
```

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Use Modello for XML report generation instead of writing reports using StringBuilder. This is less error-prone and we can easily generate the XML Schema for the reports.

This also fixes #755

### CODE REVIEW

File: src/main/mdo/dependency-updates-report.mdo [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+            </fields>
+        </class>
+
+        <class>
+            <name>DependencyReportSummary</name>
+            <fields>
+                <field>
+                    <name>usingLastVersion</name>
+                    <required>true</required>
+                    <type>String</type>
```
Comment (ajarmoniuk) [inline src/main/mdo/dependency-updates-report.mdo:79]: So, I'd really like to have an `int` there, but all serializer implementations will refrain from producing an element there if the value is equal to zero; they seem to treat zero as a non-value. I guess that might call for a change request in Modello?

Comment (ajarmoniuk) [reply src/main/mdo/dependency-updates-report.mdo:79]: Or maybe replace Modello with something else: e.g. Jackson or JAXB?

File: src/main/mdo/dependency-updates-report.mdo [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<model xmlns="http://codehaus-plexus.github.io/MODELLO/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="http://codehaus-plexus.github.io/MODELLO/2.0.0 https://codehaus-plexus.github.io/modello/xsd/modello-2.0.0.xsd"
+       xsd.namespace="http://mojo.codehaus.org/versions-maven-plugin/rule/2.0.0"
+       xsd.targetNamespace="http://mojo.codehaus.org/versions-maven-plugin/rule/2.0.0">
```
Comment (slawekjaranowski) [inline src/main/mdo/dependency-updates-report.mdo:25]: it is not schema for rules ...

Comment (ajarmoniuk) [reply src/main/mdo/dependency-updates-report.mdo]: Oops! Okay, corrected.

Comment (ajarmoniuk) [reply src/main/mdo/dependency-updates-report.mdo]: What do you think of the following?

http://mojo.codehaus.org/versions-maven-plugin/dependency-updates-report/1.0.0
http://mojo.codehaus.org/versions-maven-plugin/plugin-updates-report/1.0.0

File: src/main/mdo/dependency-updates-report.mdo [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<model xmlns="http://codehaus-plexus.github.io/MODELLO/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="http://codehaus-plexus.github.io/MODELLO/2.0.0 https://codehaus-plexus.github.io/modello/xsd/modello-2.0.0.xsd"
+       xsd.namespace="http://mojo.codehaus.org/versions-maven-plugin/dependency-updates-report/1.0.0"
+       xsd.targetNamespace="http://mojo.codehaus.org/versions-maven-plugin/dependency-updates-report/1.0.0">
```
Comment (slawekjaranowski) [inline src/main/mdo/dependency-updates-report.mdo:25]: What do you think about namespase like:

```
https://www.mojohaus.org/VERSIONS/DEPENDENCY-UPDATES-REPORT/1.0.0
```

similar to Maven xsd:
http://maven.apache.org/xsd/

File: src/main/mdo/dependency-updates-report.mdo (hunk 1 above)
Comment (slawekjaranowski) [reply src/main/mdo/dependency-updates-report.mdo:79]: I would like to stay with modelo, not introduce too much idea in one time

**Pull request merged**